pkg/security/security.go

```diff
@@ -140,6 +141,12 @@ type Options struct {
 
 	// Name of the Service Account
 	ServiceAccount string
+
+	// Enable token fetch and exchange for XDS
+	XdsEnableTokenFetchExchange bool
+
+	// Token manager for the token exchange of XDS
+	TokenManager stsservice.TokenManager
```

Comment:
I don't like this dependency - too much spaghetti. 
pkg should not depend on packages and bring deps. 

Can you maybe define an interface here, which will also allow other implementations ?

Reply:
If not refer to stsservice.TokenManager, we need to define an interface containing GenerateToken() in security.go, which means StsRequestParameters also needs to be defined in security.go. Defining them again in security.go instead of referring to the existing interface and structures will cause a lot of duplicate code.
type TokenManager interface {
	GenerateToken(parameters StsRequestParameters) ([]byte, error)
         ...
}

Reply:
Move the parameters here. Again, we should not have a dependency to the GCP impl.

Reply:
Actually - there is only one call to TokenManager, in the TokenSource. 
Can you make this field "google.golang.org/grpc/credentials.PerRPCCredentials ( i.e. use the interface defined by gRPC ), and use a small stub implementing the interface ? The TokenMAnager and StsRequestParameters are not a very good fit for what we need - STS is a particular implementation of PerRPCCredentials, and the struct/impl are geared toward implementing an STS server, not client.

Reply:
TokenProvider, which is taking in a TokenExchanger, and now a TokenManager, does implement PerRPCCredentials. Confused by all the TokenXYZ? I am as well, which is why I am really hoping https://github.com/istio/istio/issues/29753 can be done.

Reply:
The dependency has been removed by following the comment of moving the definition of StsRequestParameters and TokenManager to this file. Other refactoring work should be addressed in a separate PR.